src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -2,48 +2,36 @@
 
 import nextstep.qna.CannotDeleteException;
 import nextstep.qna.NotFoundException;
-import nextstep.qna.domain.*;
+import nextstep.qna.domain.answer.AnswerRepository;
+import nextstep.qna.domain.deleteHistory.DeleteHistory;
+import nextstep.qna.domain.question.Question;
+import nextstep.qna.domain.question.QuestionRepository;
 import nextstep.users.domain.NsUser;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import javax.annotation.Resource;
-import java.time.LocalDateTime;
-import java.util.ArrayList;
 import java.util.List;
 
 @Service("qnaService")
 public class QnAService {
-    @Resource(name = "questionRepository")
-    private QuestionRepository questionRepository;
-
-    @Resource(name = "answerRepository")
-    private AnswerRepository answerRepository;
-
-    @Resource(name = "deleteHistoryService")
-    private DeleteHistoryService deleteHistoryService;
-
-    @Transactional
-    public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
-        Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
-    }
+  @Resource(name = "questionRepository")
+  private QuestionRepository questionRepository;
+
+  @Resource(name = "answerRepository")
+  private AnswerRepository answerRepository;
+
+  @Resource(name = "deleteHistoryService")
+  private DeleteHistoryService deleteHistoryService;
+
+  @Transactional
+  public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
+
+    // 조회
+    Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
+
+    // 삭제 & 히스토리 저장
+    List<DeleteHistory> histories = question.delete(loginUser);
+    deleteHistoryService.saveAll(histories);
```

Comment:
```suggestion
    Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
    List<DeleteHistory> histories = question.delete(loginUser);
    deleteHistoryService.saveAll(histories);
```
주석은 제거해도 좋겠네요. 이미 메소드 이름만으로도 의도가 잘 나타나고 있어서 충분합니다 👍